var/spack/repos/builtin/packages/libzmq/package.py

```diff
@@ -12,110 +14,229 @@ class Libzmq(AutotoolsPackage):
-        config_args.extend(self.enable_or_disable("libunwind"))
+        args = self.with_or_without("docs")
+        args += self.enable_or_disable("drafts")
+        args += self.enable_or_disable("libbsd")
+        args += self.with_or_without("libsodium")
+        args += self.enable_or_disable("libunwind")
         # the package won't compile with newer compilers because warnings
         # are converted to errors. Hence, disable such conversion.
-        # this option was only added in version 4.2.3.
         if self.spec.version >= Version("4.2.3"):
-            config_args.append("--disable-Werror")
-        if "clang" in self.compiler.cc:
-            config_args.append("CFLAGS=-Wno-gnu")
-            config_args.append("CXXFLAGS=-Wno-gnu")
-        return config_args
+            args.append("--disable-Werror")
+        if self.spec.compiler.name == "clang":
+            args.append("CFLAGS=-Wno-gnu")
+            args.append("CXXFLAGS=-Wno-gnu")
+        return args
+
+    @run_after("configure")
+    @on_package_attributes(run_tests=True)
+    def check_configure(self):
+        """sanity checks to avoid mistakes like in https://github.com/spack/spack/issues/40455"""
+        config_log = open("config.log", "r").read()
+        macros = []
+        if self.spec.satisfies("+drafts"):
+            macros.append("#define ZMQ_BUILD_DRAFT_API 1")
+        if self.spec.satisfies("+libbsd"):
+            macros.append("#define ZMQ_HAVE_LIBBSD 1")
+        if self.spec.satisfies("+libsodium"):
+            macros.append("#define ZMQ_USE_LIBSODIUM 1")
+        if self.spec.satisfies("+libunwind"):
+            macros.append("#define HAVE_LIBUNWIND 1")
+        # https://github.com/zeromq/libzmq/commit/ff47aeb791e134a78bc386e12eea67618e0bf2f7
+        # Since 4.3.5 the curve feature is only available via libsodium backend
+        if not self.spec.satisfies("@4.3.5:~libsodium"):
+            macros.append("#define ZMQ_HAVE_CURVE 1")
+        for macro in macros:
+            if not re.search(macro, config_log):
+                raise RuntimeError(f"Expected '{macro}' in config.log")
+
+    sanity_check_is_file = [
+        join_path("include", "zmq.h"),
+        join_path("lib", "libzmq.so"),
+        join_path("lib", "pkgconfig", "libzmq.pc"),
+    ]
+
+    @run_after("install")
+    @on_package_attributes(run_tests=True)
+    def check_presence_of_files(self):
+        """check presence of important (conditional) files"""
+        if self.spec.satisfies("+docs"):
+            assert os.path.exists(join_path(self.prefix.share.man.man7, "zmq.7"))
+
+    test_requires_compiler = True
+
+    def _compiled_executable(self, source_code):
+        """compile the given C source code as executable"""
+        exe = f"test_{abs(hash(source_code)):08x}"
+        source_file = f"{exe}.c"
+        with open(source_file, "w") as file:
+            file.write(source_code)
+        which(env["CC"])(
```

Comment:
I have seen many times `Executable(self.compiler.cc)` instead of this `which(env["CC"])`.

The `self.compiler.cc` looks more like "spack" to me.

Should this be used instead?

Reply:
> The self.compiler.cc looks more like "spack" to me.

I like it better as well, changed it.

Reply:
Thanks for using `Executable` and simplifying the file writing. Resolving.